### ISSUE TITLE
HTStream as a Singularity/Docker container

### ISSUE THREAD

**Issue by germaine260** [ibest/HTStream#162]
Hi I'm trying to use HTStream on my school's computational cluster, which accommodates usage of software not installed on the system (i.e. HTStream) via Linux containers. Will it be possible to make HTStream available as a Singularity or Docker container that I can pull down? I tried building my own container unsuccessfully and will greatly appreciate your help, or any advice on how I can get HTStream to work. Thanks!

**Comment by dstreett** [comment]
Hi @germaine260, thanks for the note. A dockerfile/image is a good idea and has been talked about quite a bit. This is a good excuse to get one setup.

I can get something spun up for you tonight.

**Comment by germaine260** [comment]
Thanks David! Appreciate your help.

On Sep 9, 2019, at 6:19 AM, David Streett <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @germaine260<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_germaine260&d=DwMCaQ&c=iORugZls2LlYyCAZRB3XLg&r=82HCZbegW7rprjtWFWMDWMvuhoVi-dFrimLm-q4Lu9E&m=QhdI87-iP0MWU90GsaNSX6wZRqPu9UZf3qDTQtc9tWA&s=mVWMchzagqnBSWk5FpJnzQO1-NODHLI3abukA_YGxJc&e=>, thanks for the note. A dockerfile/image is a good idea and has been talked about quite a bit. This is a good excuse to get one setup.

I can get something spun up for you tonight.

**Comment by germaine260** [comment]
Hi David,

Any updates on generating a dockerfile/image for HTStream for use on my school’s HPC? I’ve been trying to get to HTStream via a conda virtual environment but has been unsuccessful. Thanks for your help.

-Germaine

On Sep 9, 2019, at 10:40 AM, Yong, Germaine <Germaine.Yong@ucsf.edu<mailto:Germaine.Yong@ucsf.edu>> wrote:

Thanks David! Appreciate your help.

On Sep 9, 2019, at 6:19 AM, David Streett <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @germaine260<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_germaine260&d=DwMCaQ&c=iORugZls2LlYyCAZRB3XLg&r=82HCZbegW7rprjtWFWMDWMvuhoVi-dFrimLm-q4Lu9E&m=QhdI87-iP0MWU90GsaNSX6wZRqPu9UZf3qDTQtc9tWA&s=mVWMchzagqnBSWk5FpJnzQO1-NODHLI3abukA_YGxJc&e=>, thanks for the note. A dockerfile/image is a good idea and has been talked about quite a bit. This is a good excuse to get one setup.

I can get something spun up for you tonight.

**Comment by dstreett** [comment]
Sorry about the delay. Just waiting on someone to approve the PR. https://github.com/ibest/HTStream/pull/163 You are welcome to test and approve the PR if it satisfies your needs!

**Comment by germaine260** [comment]
Thanks for this David. My school’s HPC uses Singularity to access Docker containers — I’m trying to follow instructions on this page (https://ucsf-hpc.github.io/wynton/software/singularity.html) to build a container using your htstream Docker container, but I can’t find the htstream container on the docker repository (https://hub.docker.com/search?q=htstream&type=image).

I will be glad to test and approve your PR but I do not know how to do so; I am an amateur at Github and HPC clusters (Sorry!). After reading up a bit, might it work if you can push the Docker image to the Github Package Registry (https://help.github.com/en/articles/configuring-docker-for-use-with-github-package-registry#installing-a-package) or make it available on the Docker Hub?

Thanks for your help!
-Germaine

On Sep 17, 2019, at 7:18 AM, David Streett <notifications@github.com<mailto:notifications@github.com>> wrote:


Sorry about the delay. Just waiting on someone to approve the PR. #163<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ibest_HTStream_pull_163&d=DwMCaQ&c=iORugZls2LlYyCAZRB3XLg&r=82HCZbegW7rprjtWFWMDWMvuhoVi-dFrimLm-q4Lu9E&m=U2QVhlluTH6rKJ26Hv5RZeeMyomwL3XqeRcgCIPVz7U&s=KVDg7vySNiRGeg3kkDGM0WWq5PIFGka8L72hw056IQ8&e=> You are welcome to test and approve the PR if it satisfies your needs!

**Comment by dstreett** [comment]
No worries! Talked to the other devs and they are happy with the dockerfile being merged in.

I will upload the image tonight now that the code has been approved.

**Comment by dstreett** [comment]
Sorry about the delay!

`docker pull dzs74/htstream` is now working. Here is the location on docker hub -> https://hub.docker.com/r/dzs74/htstream .

Let me know if you run into any issues.

**Comment by germaine260** [comment]
Thanks David, the dockerfile worked perfectly!

**Comment by dstreett** [comment]
Glad it worked! I'm going to close this issue, but If you have any more questions - always feel free to open a new issue.
